template/src/core/app/appStateReducer.ts

```diff
@@ -0,0 +1,30 @@
+const initialAppState = {
+  hasLoggedIn: false
```

Comment:
The more realistic example would have been to have an actual `account` property
```suggestion
  account: null as null | Record<string, any>
```

Then we can rename `ProtectedRoute` to `RequireAccount` which is a better name for it.